feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-multi-selection-base.component.ts

```diff
@@ -0,0 +1,90 @@
+import { ConfiguratorAttributeBaseComponent } from './configurator-attribute-base.component';
+import { Directive, EventEmitter, Input, Output } from '@angular/core';
+import { Configurator } from '../../../../core/model/configurator.model';
+import { ConfigFormUpdateEvent } from '../../../form/configurator-form.event';
+import { ConfiguratorAttributeQuantityComponentOptions } from '../../quantity/configurator-attribute-quantity.component';
+import { map } from 'rxjs/operators';
+import { BehaviorSubject } from 'rxjs';
+import { ConfiguratorAttributeQuantityService } from '../../quantity/configurator-attribute-quantity.service';
+
+@Directive()
+// eslint-disable-next-line @angular-eslint/directive-class-suffix
+export abstract class ConfiguratorAttributeMultiSelectionBaseComponent extends ConfiguratorAttributeBaseComponent {
+  loading$ = new BehaviorSubject<boolean>(false);
+
+  @Input() attribute: Configurator.Attribute;
+  @Input() ownerKey: string;
+  @Output() selectionChange = new EventEmitter<ConfigFormUpdateEvent>();
+
+  // TODO(#11681): make quantityService a required dependency
+  constructor(
+    protected quantityService?: ConfiguratorAttributeQuantityService
+  ) {
+    super();
```

Comment:
In single base component, the constructor breaking change is handled in parent, and in this component, it is handled in children. Can we make them consistent?

Reply:
@AlexanderKuschner, let us have a look at it?

Reply:
I check the codes again, the breaking changes have to handled in children, because some children are new classes, they don't have any breaking change issues.

Reply:
The breaking changes handling in single select is moved to child components.